platform/engineering/design-docs/2020-05-26-content-validation.md

```diff
@@ -0,0 +1,227 @@
+
+## Specifics
+
+### Detailed Design
+
+Using the current content build script as a reference, we will write another build script that only includes the steps necessary for the content validation. We will reuse the existing broken link check from the current content build. And using the existing accessibility tests as a reference, we will configure the accessibility tests to not stop on failure so we can identify all the accessibility errors. The script will generate a report of all the broken links and all the accessibility errors. Those reports will be sent to the #cms-team channel, just like the current broken link notifications today.
+
+Lastly, we will write a scheduled Jenkins workflow to run the content validation script every workday at midnight.
+
+### Code Location
+
+The content validation script and scheduled job config will live in the [new `content-build` repo](https://github.com/department-of-veterans-affairs/content-build).
+
+### Testing Plan
+
+1. Once the content validation script is written, we can run the script locally.
+2. Once the script is behaving correctly in local environment, we can call the script in a scheduled job.
+3. Once the script is being called in the scheduled job, we can monitor and optimize it.
+
+### Logging
+
+Currently, invalid content is logged in the build log during staging deploys. And failed builds trigger generic [Slack notifications](https://dsva.slack.com/archives/C37M86Y8G/p1591099447374800).
+
+The updated content validation approach will continue logging invalid content errors in the build log, and will also post an automated report of the invalid content in the #cms-team Slack channel.
+
+### Debugging
+
+##### Broken links
+
+1. Confirm that the content build is passing
+2. Modify an HTML file, like [`header.html`](https://github.com/department-of-veterans-affairs/content-build/blob/master/src/site/includes/header.html), to have a broken link
+3. Run the build again to confirm that it is failing
+
+##### Accessibility errors
+
+1. Confirm that the content build is passing
+2. Modify an HTML file, like [`header.html`](https://github.com/department-of-veterans-affairs/content-build/blob/master/src/site/includes/header.html), to have an accessibility error
+3. Run the build again to confirm that it is failing
+
+### Caveats
+
+- The current broken link checking in Drupal can handle redirects, but those redirects are not currently synchronized with `vets-website`. When the redirects are not synchronized between `vets-website` and Drupal, a page that is moved in Drupal may pass the Drupal broken link check but fail the `vets-website` broken link check.
```

Comment:
I don't understand why this is a caveat. anchor tags from VA's CMS shouldn't point to redirects- redirects are for external referrers i.e. content we don't control.

Reply:
I added this based on [a comment from Steve W](https://github.com/department-of-veterans-affairs/va.gov-team/pull/9658#discussion_r434864015). 

Do you think this paragraph needs to be moved (perhaps to the **Background** section), is incorrect, is confusing, and/or something else?

Reply:
Re: whether or not this caveat needs to be here

Does this affect your design doc's plans in any way? It's true information, but I don't think it's relevant to the changes you're proposing in this doc.

Reply:
It doesn't affect the design doc's plans in any way. I've moved it from caveats to background in https://github.com/department-of-veterans-affairs/va.gov-team/pull/9658/commits/4efc42728ba68c5c467fded098d3791f73d635fa.